woocommerce/woocommerce
# Hadren styles for interactive elements in Checkout block

This PR switches some of our interactive elements in Checkout to use an Ariakit Button element (rendered as div), this prevents breakage by aggressive styles coming from theme.

The PR changes:

- Panel titles (Order Summary, Add a coupon).
- Shipping vs Local Pickup Selectors.
- Address line 2 toggle.
- Edit address toggle.

Closes https://github.com/woocommerce/woocommerce/issues/50954

For the purpose of this PR, just make sure elements aren't broken and look the same.

## Screenshots:

### Core theme with no changes:

#### Before:

<img width="1353" alt="Screenshot 2024-09-13 at 11 33 30 AM" src="https://github.com/user-attachments/assets/138e848d-78f8-411f-a76f-4cf15edc6fd7">
<img width="1351" alt="Screenshot 2024-09-13 at 11 33 12 AM" src="https://github.com/user-attachments/assets/94f38346-ae51-4408-b136-97f8586b853b">

#### After:

<img width="1394" alt="image" src="https://github.com/user-attachments/assets/41fae2f0-f2a5-4502-a1d2-28bddd05dfb5">
<img width="1348" alt="Screenshot 2024-09-13 at 12 04 32 PM" src="https://github.com/user-attachments/assets/80ee6169-5881-4dce-857c-cfafdb45713a">


### Third party theme with changes:

#### Before:

<img width="1187" alt="Screenshot 2024-09-12 at 5 43 35 PM" src="https://github.com/user-attachments/assets/1e0c89d5-00ce-442b-96a5-6d49c80b3f44">

#### After:

<img width="1220" alt="Screenshot 2024-09-12 at 6 48 05 PM" src="https://github.com/user-attachments/assets/384d8518-372a-4316-81e2-fc90c1b5accc">



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Select a theme (storefront) and test panels titles, local pickup vs shipping selectors, address line 2 toggle, and address edit button.
2. Make sure they're interactive (cursor is pointer).
3. Make sure you can select them with keyboard and focus them.
4. Make sure you can toggle them using keyboard.
5. Go the Checkout edit page, select local pickup, enable price and disable icon.
6. Visit frontend, make sure they're shown on separate lines.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Harden styles for interactive elements in Checkout block to prevent style leakage.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
